proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,237 @@
+# Unlock Existentials with Fixed Associated Types
+
+* Proposal: [SE-NNNN](NNNN-unlock-existentials-with-fixed-associated-types.md)
+* Authors: [Filip Sakel](https://github.com/filip-sakel), [Anthony Latsis](https://github.com/AnthonyLatsis)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Today, Swift protocols are divided into two categories: those that _can_ be used as fully-fledged types, and those that can only be used as generic constraints because they have `Self` or associated type requirements. However, in some cases where a protocol inherits and fixes the associated types of another protocol - that doesn't have `Self` requirements, this constraint seems rather unintuitive. This proposal aims to relax this needless contraint allowing such protocols to be used as Types.
```

Comment:
Could you surround the error message in `ticks`? The error is already confusing and inaccurate enough, we would not want people to read it as ordinary text.

Reply:
I don't think that'd be wrong. I think we should either find a satisfactory definition that doesn't require ticks (and maybe reference the recognizable error in parenthesis with the ticks) or just not include the ticks. I say that because the introduction should be plain text and include a proper definition for what's to be discussed.

Reply:
The problem is that a real definition is not as simple, but we shouldn't need to clarify the exact definition of which existential types are not supported today (being relative is enough for this proposal). While the plain text makes for an awful explanation, a *quote* completes the explanation by having everyone recognize the error message.

> I say that because the introduction should be plain text

There is no such rule or convention for SE proposals.

Reply:
> There is no such rule or convention for SE proposals.

I'm aware of that. I am, however, concerned about how understandable that will be to future readers. Reading older proposal is quite common; therefore, if we opt to use an error message relevant to the current diagnostics, this proposal might be confusing to future readers. That's why I propose finding a short - yet elegant - definition that will make sense even without knowledge of the current diagnostics' error messages (I'm not saying that finding such a definition is easy).

Of course, this the current message is not absolutely terrible; it still makes sense, even without being completely accurate or easily graspable. Thus, future readers, without necessarily understanding that this used to be an error message, would still be able to understand what we are referring to. So, I've added the ticks, but let me know what you think:

> Today, Swift protocols are divided into two categories: those that can be used as fully-fledged types, and those that can only be used as generic constraints because they have 'Self' or associated type requirements. However, in some cases where a protocol inherits and fixes the associated types of another protocol - that doesn't have Self requirements, this constraint seems rather unintuitive. This proposal aims to relax this needless contraint allowing such protocols to be used as Types.